portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,32 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+function popfunc(){
+    shown.pop()
+}
 /**
- * Adds a random greeting to the page.
+ * gets a random celebration picture.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+var shown = []
+function getRandomCelebration() {
+    var imageIndex = Math.floor(Math.random() * 10) + 1;
+    if (shown.length < 11){ 
+        if (shown.includes(imageIndex)){
+            while( shown.includes(imageIndex)){
+                imageIndex = Math.floor(Math.random() * 10) + 1
+            }
+        }
+    }else{
+        shown.forEach(popfunc)
```

Comment:
Uhh... I'm a Javascript novice, so I didn't even know that there's a `forEach()` method on a JS array (or list?), but wondering what `popfunc` here is? Is it a built-in function? Also the logic here seems not very easy to understand, and it might help to add some comments about what you are trying to do. No need to follow up on this comment right away, and let's talk about this off of this review.